internal/icinga2/client_api.go

```diff
@@ -19,6 +19,25 @@ import (
 
 // This file contains Icinga 2 API related methods.
 
+// checkHTTPResponseStatusCode compares an HTTP response status code against supported Icinga 2 API codes.
+//
+// An error will be returned for invalid status codes according to the documentation. Undefined status codes are going
+// to result in errors as well.
+//
+// https://icinga.com/docs/icinga-2/latest/doc/12-icinga2-api/#http-statuses
+func checkHTTPResponseStatusCode(statusCode int, status string) error {
+	switch {
+	case 200 <= statusCode && statusCode <= 299:
+		return nil
+	case 400 <= statusCode && statusCode <= 499:
+		return fmt.Errorf("invalid HTTP request: %d, %q", statusCode, status)
+	case 500 <= statusCode && statusCode <= 599:
+		return fmt.Errorf("server HTTP error: %d, %q", statusCode, status)
+	default:
+		return fmt.Errorf("unknown HTTP error: %d, %q", statusCode, status)
```

Comment:
It looks like the HTTP error code is already included in the status message, as documented (`http.Response.Status string // e.g. "200 OK"`), so you don't need to manually add it to the error message.

```bash
2024-06-03T09:22:35.039+0200	ERROR	icinga2	Establishing an Event Stream API connection failed with an HTTP error, will be retried	{"source-id": 1, "error": "invalid HTTP request: 401, \"401 Unauthorized\"", "delay": "1s"}
2024-06-03T09:22:36.096+0200	ERROR	icinga2	Establishing an Event Stream API connection failed with an HTTP error, will be retried	{"source-id": 1, "error": "invalid HTTP request: 401, \"401 Unauthorized\"", "delay": "2s"}
---
2024-06-03T09:23:04.190+0200	ERROR	icinga2	Establishing an Event Stream API connection failed with an HTTP error, will be retried	{"source-id": 1, "error": "invalid HTTP request: 404, \"404 Not Found\"", "delay": "4s"}
2024-06-03T09:23:08.252+0200	ERROR	icinga2	Establishing an Event Stream API connection failed with an HTTP error, will be retried	{"source-id": 1, "error": "invalid HTTP request: 404, \"404 Not Found\"", "delay": "8s"}
```

Reply:
I was afraid that it might get redacted anyhow, but it seems like this is impossible.
https://cs.opensource.google/go/go/+/refs/tags/go1.22.3:src/net/http/response.go;l=168-173